lib/rex/proto/mssql/client.rb

```diff
@@ -71,6 +71,57 @@ def initialize(framework_module, framework, rhost, rport = 1433, proxies = nil)
           @current_database = ''
         end
 
+        # MS SQL Server only supports Windows and Linux
+        def map_compile_os_to_platform(server_info)
+          return Msf::Platform::Unknown.realname if server_info.blank?
+
+          os_data = server_info.downcase.encode(::Encoding::BINARY)
+
+          if os_data.match?('linux')
+            platform = Msf::Platform::Linux
+          elsif os_data.match?('windows')
+            platform = Msf::Platform::Windows
+          elsif os_data.match?('win')
+            platform = Msf::Platform::Windows
+          else
+            platform = Msf::Platform::Unknown
```

Comment:
I think instead of `unknown` we'll just default to the string that the server gives back. We'll need to update the mysql logic here for consistency